### PR TITLE
TransceiverRAD1 radioVector access optimization

### PR DESCRIPTION
**Pull request by kkroo**
WIP..

This implements a blocking `get` for `VectorFIFO` rather than spinning cycles checking if the non-blocking `get` has something for us

### CODE REVIEW

**Comment by kkroo** [comment]
CPU utilization seems to have dropped a good amount while idle. ~260% -> ~70%

**Comment by kkroo** [comment]
PTAL

**Comment by shaddi** [comment]
lgtm

**Comment by kheimerl** [comment]
I'd like a bit more testing; what was the output from this?

**Comment by kheimerl** [comment]
Utilization dropped, but not that much, and that's it?

**Comment by shaddi** [comment]
The testing I saw showed it performing about the same as before (so no huge improvements) but CPU utilization dropped dramatically.

**Comment by kheimerl** [comment]
I'd like an a/b on data rates proving that.

**Comment by kheimerl** [comment]
This is now failing on the test box: transceiver.old works but transceiver.omar does not.

File: TransceiverRAD1/Transceiver.cpp [hunk 1]
```diff
@@ -989,16 +989,10 @@ void Demodulator::driveDemod(bool wSingleARFCN)
...
-      RadioClock *radioClock = (mRadioInterface->getClock());
-      //radioClock->wait();
-      demodBurst = mDemodFIFO->get();
-    }
-  }
-  else {
-    if (!demodBurst) return;
+  demodBurst = mDemodFIFO->get(true);
+  if (!demodBurst) {
+    LOG(ERR) << "Failed to get radioVector";
```
Comment (shaddi) [inline TransceiverRAD1/Transceiver.cpp]: This is a common case, we should remove this before merging this in.

**Comment by kkroo** [comment]
It seems that this is causing deadlocks, friends.

**Comment by kheimerl** [comment]
well that's not good. Track down the deadlock as before?

**Comment by kkroo** [comment]
Hunting for it.. Things seem to crash after calling `get`

```
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:24:put: Trying to get lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:26:put: Got lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:29:put: Released lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:24:put: Trying to get lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:26:put: Got lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:29:put: Released lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:24:put: Trying to get lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:26:put: Got lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:29:put: Released lock in put
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 Transceiver.cpp:992:driveDemod: Tring to get demodBurst
Jul 20 22:19:20 endaga-demo2 transceiver: ERR 3074:3095 2015-07-20T22:19:20.8 radioInterface.cpp:38:get: Got lock in get(true)
Jul 20 22:19:27 endaga-demo2 openbts: ERR 3080:3080 2015-07-20T22:19:27.0 iputils.cpp:242:runcmd: sub-process did not complete in 5 sec
```

**Comment by kheimerl** [comment]
You can set logging for Mutex.cpp and it'll log every lock taking and releasing. then it's pretty simple to find who grabbed it but didn't let it go.

**Comment by kkroo** [comment]
the deadlock seems to be fixed.. lemme know if you run into more weirdness

**Comment by kkroo** [comment]
@kheimerl says LGTM

**Pull request merged**